test/interactions/clickInteractionTests.ts

```diff
@@ -1,158 +1,208 @@
+          const point = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+
+          clickPoint(point, mode);
+
+          assert.isTrue(callback.getCalled(), "callback called on clicking Component without moving mouse");
+          assert.deepEqual(callback.getLastPoint(), point, "was passed correct point");
+        });
+
+        it("provides correct mouseup point to onClick callback on click for " + TestMethods.InteractionMode[mode], () => {
+          const startPoint = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+          const endPoint = {x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isTrue(callback.getCalled(), "callback called on clicking and releasing inside the Component");
+          assert.deepEqual(callback.getLastPoint(), endPoint, "was passed mouseup point");
+        });
+
+        it("does not invoke callback if click is released outside Component for " + TestMethods.InteractionMode[mode], () => {
+          const startPoint = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+          const endPoint = {x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isFalse(callback.getCalled(), "callback not called if mouse is released outside Component");
+        });
+
+        it("does not invoke callback if click is started outside Component for " + TestMethods.InteractionMode[mode], () => {
+          const startPoint = {x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2};
+          const endPoint = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isFalse(callback.getCalled(), "callback not called if click was started outside Component");
+        });
+
+        it("invokes callback if the mouse is moved out then back inside the Component before releasing for "
+          + TestMethods.InteractionMode[mode], () => {
+          TestMethods.triggerFakeInteractionEvent(mode,
+                                                  TestMethods.InteractionType.Start,
+                                                  component.content(),
+                                                  SVG_WIDTH / 2,
+                                                  SVG_HEIGHT / 2);
+          TestMethods.triggerFakeInteractionEvent(mode,
+                                                  TestMethods.InteractionType.Move,
+                                                  component.content(),
+                                                  SVG_WIDTH * 2,
+                                                  SVG_HEIGHT * 2);
+          TestMethods.triggerFakeInteractionEvent(mode,
+                                                  TestMethods.InteractionType.End,
+                                                  component.content(),
+                                                  SVG_WIDTH / 2,
+                                                  SVG_HEIGHT / 2);
```

Comment:
Our current indentation strategy is a little weak and not enforced (tslint does not have a defined rule for this I believe?), but I think we prefer 2 space indentation per nesting in all cases.  What do you think here?